### PR TITLE
Prettify step generator and deprecate password

### PR DESCRIPTION
**Pull request by casz**
* Deprecating username and password.
* Created descriptors and jelly configs as needed.
* Simplified mostly 👍

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -63,23 +63,8 @@
 		</pluginRepository>
 	</pluginRepositories>
 
-	<distributionManagement>
```
Comment (casz) [inline pom.xml]: Provided py parent POM

File: pom.xml [hunk 2]
```diff
@@ -5,14 +5,14 @@
...
 		<groupId>org.jenkins-ci.plugins</groupId>
 		<artifactId>plugin</artifactId>
-		<version>2.23</version>
+		<version>3.2</version>
 	</parent>
 
 	<properties>
+		<java.level>7</java.level>
 		<jenkins.version>2.7.4</jenkins.version>
-		<hpi-plugin.version>1.115</hpi-plugin.version>
```
Comment (casz) [inline pom.xml]: Let parent POM decide

File: pom.xml [hunk 3]
```diff
@@ -229,5 +185,11 @@ Changelog of Violation Comments to Bitbucket Server Plugin.
 			<version>2.5.0</version>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
```
Comment (casz) [inline pom.xml]: Parent POM started requiring upper bound dependency, this is in test scope. Jenkins is using an ancient one while PODAM is using a newer one. https://github.com/jenkinsci/plugin-pom/blob/master/CHANGELOG.md#231

File: src/main/java/org/jenkinsci/plugins/jvctb/ViolationsToBitbucketServerDescriptor.java [hunk 4]
```diff
@@ -81,51 +52,14 @@ public boolean isApplicable(
...
-      final ViolationConfig violationConfig = config.getViolationConfigs().get(i);
-      violationConfig.setPattern(pattern);
-      final String reporter = reporters.get(i);
-      violationConfig.setReporter(reporter);
-      i++;
-    }
-    final ViolationsToBitbucketServerRecorder publisher = new ViolationsToBitbucketServerRecorder();
-    publisher.setConfig(config);
-    return publisher;
+    return req.bindJSON(ViolationsToBitbucketServerRecorder.class, formData);
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/jvctb/ViolationsToBitbucketServerDescriptor.java]: Let life be simple 😅

File: src/main/java/org/jenkinsci/plugins/jvctb/ViolationsToBitbucketServerGlobalConfiguration.java [hunk 5]
```diff
@@ -37,7 +37,7 @@
   public String repoSlug;
   public String username;
   private String usernamePasswordCredentialsId;
-  private SEVERITY minSeverity;
+  private SEVERITY minSeverity = SEVERITY.INFO;
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/jvctb/ViolationsToBitbucketServerGlobalConfiguration.java]: Ensure default

File: src/main/java/org/jenkinsci/plugins/jvctb/perform/JvctbPerformer.java [hunk 6]
```diff
@@ -25,44 +51,15 @@
...
 
 public class JvctbPerformer {
   private static Logger LOG = Logger.getLogger(JvctbPerformer.class.getSimpleName());
 
   @VisibleForTesting
   public static void doPerform(
-      final ViolationsToBitbucketServerConfig config,
-      final File workspace,
-      final TaskListener listener)
+          final ViolationsToBitbucketServerConfig config,
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/jvctb/perform/JvctbPerformer.java]: ARGH fighting the two space indenting 😢

File: src/main/java/org/jenkinsci/plugins/jvctb/perform/JvctbPerformer.java [hunk 7]
```diff
@@ -25,44 +51,15 @@
...
+          final File workspace,
+          StandardUsernamePasswordCredentials standardUsernamePasswordCredentials, final TaskListener listener)
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/jvctb/perform/JvctbPerformer.java]: I haven't tested it.

File: src/main/java/org/jenkinsci/plugins/jvctb/config/ViolationsToBitbucketServerConfig.java [hunk 8]
```diff
@@ -412,7 +362,33 @@ public boolean isKeepOldComments() {
...
+
+  @Extension
+  public static class DescriptorImpl extends Descriptor<ViolationsToBitbucketServerConfig> {
+    @Nonnull
+    @Override
+    public String getDisplayName() {
+      return "Violations To Bitbucket Server Config";
+    }
+
+    public static final ListBoxModel TYPES = new ListBoxModel(
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/jvctb/config/ViolationsToBitbucketServerConfig.java]: Should switch to for each loop

File: src/main/java/org/jenkinsci/plugins/jvctb/ViolationsToBitbucketServerRecorder.java [hunk 9]
```diff
@@ -20,7 +20,7 @@
 import static org.jenkinsci.plugins.jvctb.perform.JvctbPerformer.jvctsPerform;
 
 public class ViolationsToBitbucketServerRecorder extends Recorder implements SimpleBuildStep {
-  @Extension
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/jvctb/ViolationsToBitbucketServerRecorder.java]: forgot to remove the extension in #43 from here when it was added to the actual descriptor class.

**Comment by casz** [comment]
@tomasbjerre Should be good to go once #46 is merged 👍

File: src/main/java/org/jenkinsci/plugins/jvctb/perform/JvctbPerformer.java [hunk 10]
```diff
@@ -1,8 +1,34 @@
 package org.jenkinsci.plugins.jvctb.perform;
 
+import java.io.File;
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/jvctb/perform/JvctbPerformer.java]: Whats the magic import order :|

**Comment by casz** [comment]
I would argue we should bump Jenkins Level up to 2.60.3 at least to get Java 8 target 👍 
http://stats.jenkins.io/pluginversions/violation-comments-to-stash.html

To replace some of that Guava crust 🙊

**Comment by tomasbjerre** [comment]
This is great stuff! I will have a closer look later.

For now, do an `mvn package` to have the code formatted with Google Java Format. There are a couple of minor formatting issues in `ViolationsToBitbucketServerGlobalConfiguration` and `ViolationsToBitbucketServerConfig`.

**Pull request merged**